homeassistant/components/broadlink/config_flow.py

```diff
@@ -143,6 +134,30 @@ async def async_step_user(self, user_input=None):
             errors=errors,
         )
 
+    async def async_step_manual(self, user_input=None):
+        """Configure the device manually."""
+        device = self.device
+        errors = {}
+
+        if user_input is None:
+            data_schema = {
+                vol.Required(CONF_DEVICE_CLASS): vol.In(sorted(TYPES_AND_CLASSES)),
```

Comment:
Is the thought they should try each one until something works ?

Reply:
Yes. If it doesn't work, they need to remove the device and set it up again.

Reply:
Would it be reasonable to try each update method in the list instead of making the user work through each one?

Reply:
It does not work, as some devices have the same update method, but the other methods are different. How to differentiate an RM4 mini (IR only) from an RM4 pro (IR/RF), for example?

This can also lead to unwanted behavior. In RM pro, packets starting with `b'\x04\x00'` mean "get code from buffer". In RM4 pro, the same thing is used to prefix a packet with 4 bytes.